sshd/zenith/sshd/backends/crd.py

```diff
@@ -0,0 +1,182 @@
+        except ApiError as exc:
+            if exc.status_code == 404:
+                # Fetch the service so it can own the endpoints resource
+                service = ekservices.fetch(subdomain)
+                endpoints = ekendpoints.create(
+                    {
+                        "metadata": {
+                            "name": service["metadata"]["name"],
+                            "namespace": service["metadata"]["namespace"],
+                            "ownerReferences": [
+                                {
+                                    "apiVersion": service["apiVersion"],
+                                    "kind": service["kind"],
+                                    "name": service["metadata"]["name"],
+                                    "uid": service["metadata"]["uid"],
+                                    "blockOwnerDeletion": True,
+                                },
+                            ],
+                        }
+                    }
+                )
+            else:
+                raise
+
+        # Create a lease for the tunnel
+        ekleases = self.ekclient.api(self.api_version).resource("leases")
+        lease = ekleases.create(
+            {
+                "metadata": {
+                    # Generate a name based on the subdomain
+                    # We will use this as the tunnel ID
+                    "generateName": f"{subdomain}-",
+                    "ownerReferences": [
+                        {
+                            "apiVersion": endpoints["apiVersion"],
+                            "kind": endpoints["kind"],
+                            "name": endpoints["metadata"]["name"],
+                            "uid": endpoints["metadata"]["uid"],
+                            "blockOwnerDeletion": True,
+                        },
+                    ],
+                },
+                "spec": {
+                    "renewedAt": isotime(),
+                    "ttl": ttl,
+                    "reapAfter": reap_after,
+                }
+            }
+        )
+
+        # The tunnel ID is the lease name with the subdomain prefix removed
+        tunnel_id = lease["metadata"]["name"].removeprefix(f"{subdomain}-")
+
+        # Update the endpoints resource with the endpoint definition
+        # The endpoints resource should already exist
+        ekendpoints.patch(
+            subdomain,
+            {
+                "spec": {
+                    "endpoints": {
+                        tunnel_id: {
+                            "address": host,
+                            "port": port,
+                            # The initial status is critical, until the first heartbeat
+                            "status": base.TunnelStatus.CRITICAL.value,
+                            "config": config_dict,
+                        },
+                    },
+                },
+            }
+        )
+    
+        return tunnel_id
```

Comment:
Feels like this needs a nice log message? tunnel setup, or similar? I guess its covered by the caller maybe?

Maybe just debug logs to help if we hit issues? I guess we get those from easykube in that case/

Reply:
easykube logs all the API requests